src/inmanta/agent/handler.py

```diff
@@ -551,28 +551,49 @@ def _should_reload() -> bool:
                 raise Exception(f"Failed to determine whether resource should reload{error_msg_from_server}")
             return result.result["data"]
 
-        def filter_resources_in_unexpected_state(
-            reqs: abc.Mapping[ResourceIdStr, ResourceState]
+        def filter_resources_by_state(
+            reqs: abc.Mapping[ResourceIdStr, ResourceState], states: typing.Set[const.ResourceState]
         ) -> abc.Mapping[ResourceIdStr, ResourceState]:
             """
-            Return a sub-dictionary of reqs with only those resources that are in an unexpected state.
+            Return a sub-dictionary of dependencies of this resource.
+            Only keeping dependencies that are in a state that is in the provided set.
+
+            :param reqs: The list of requirements of this resource
+            :param states: The list of states that we want to keep in this sub-dictionary
             """
-            unexpected_states = {
-                const.ResourceState.available,
+
+            return {rid: state for rid, state in reqs.items() if state in states}
+
+        # Check if any dependencies got into any unexpected state
+        dependencies_in_unexpected_state = filter_resources_by_state(
+            requires,
+            {
                 const.ResourceState.dry,
                 const.ResourceState.undefined,
                 const.ResourceState.skipped_for_undefined,
-                const.ResourceState.deploying,
-            }
-            return {rid: state for rid, state in reqs.items() if state in unexpected_states}
-
-        resources_in_unexpected_state = filter_resources_in_unexpected_state(requires)
-        if resources_in_unexpected_state:
+            },
+        )
+        if dependencies_in_unexpected_state:
             ctx.set_resource_state(const.HandlerResourceState.skipped_for_dependency)
             ctx.warning(
                 "Resource %(resource)s skipped because a dependency is in an unexpected state: %(unexpected_states)s",
                 resource=resource.id.resource_version_str(),
-                unexpected_states=str({rid: state.value for rid, state in resources_in_unexpected_state.items()}),
+                unexpected_states=str({rid: state.value for rid, state in dependencies_in_unexpected_state.items()}),
+            )
+            return
+
+        # Check if any dependencies got a new desired state while this resource was waiting to deploy
+        dependencies_waiting_to_be_deployed = filter_resources_by_state(
+            requires, {const.ResourceState.available, const.ResourceState.deploying}
+        )
+        if dependencies_waiting_to_be_deployed:
+            ctx.set_resource_state(const.HandlerResourceState.skipped_for_dependency)
+            ctx.debug(
```

Comment:
Why debug? Wouldn't it be better to use `info()` here?

Reply:
I was on the fence between these 2 log types. I saw that we indicate the start of the run/deploy in `DEBUG`  ( Based on this screenshot from Guillaume) so I figured it should be debug instead of info. How do you differentiate between info and debug?
![image](https://github.com/user-attachments/assets/7191f8fe-3e5e-40fc-8239-e971c58359c1)